vdyp-common/src/main/java/ca/bc/gov/nrs/vdyp/io/parse/coe/base/BecZoneBySpeciesCoefficientParser.java

```diff
@@ -83,16 +92,20 @@ public MatrixMap2<String, String, Coefficients> parse(InputStream is, Map<String
 			var specIt = sp0Aliases.iterator();
 
 			int coefficientIndex = 0;
+			var first = new HoldFirst<Float>();
 			while (specIt.hasNext()) {
-
-				Coefficients coefficients = r.get(becZoneId, specIt.next());
+				var spec = specIt.next();
+				Coefficients coefficients = r.get(becZoneId, spec);
 				Float coe = specCoefficients.get(coefficientIndex);
-				coefficients.setCoe(index, coe);
 
-				if (valueAppliesToAllSpecies)
-					break;
+				if (valueAppliesToAllSpecies) {
+					coefficients.setCoe(index, valueShared(coe, first.get()));
+				} else {
+					coefficients.setCoe(index, value(coe, first.get()));
+				}
 
 				coefficientIndex += 1;
+				first.set(coe);
```

Comment:
This is deceiving - it looks at first glance that this is set each iteration. But of course it's a HoldFirst, meaning all but the first set are ignored. And this is all to avoid an if statement? I don't see the value.